packages/wallet-stamper/src/types.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @typedef {Object} TStamp
+ * @property {'X-Stamp'} stampHeaderName - The name of the stamp header.
+ * @property {string} stampHeaderValue - The value of the stamp header.
+ */
+export type TStamp = {
+  stampHeaderName: "X-Stamp";
+  stampHeaderValue: string;
+};
+
+/**
+ * @interface TStamper
+ * @property {function(string): Promise<TStamp>} stamp - Function to stamp an input string.
+ */
+export interface TStamper {
+  stamp: (input: string) => Promise<TStamp>;
+}
+
+/**
+ * Base interface for wallet functionalities common across different blockchain technologies.
```

Comment:
Tiny nit: I think we can use slightly more casual language, like "chains" or "ecosystems" etc :)